lib/src/geo/crs.dart

```diff
@@ -303,156 +309,144 @@ class Proj4Crs extends Crs {
   }
 
   /// returns Transformation object based on zoom
-  Transformation _getTransformationByZoom(double zoom) {
-    final transformations = _transformations;
-    if (transformations == null || transformations.isEmpty) {
-      return transformation;
-    }
-
+  _Transformation _getTransformationByZoom(double zoom) {
     final iZoom = zoom.round();
-    final lastIdx = transformations.length - 1;
-
-    return transformations[iZoom > lastIdx ? lastIdx : iZoom];
+    final lastIdx = _transformations.length - 1;
+    return _transformations[iZoom > lastIdx ? lastIdx : iZoom];
   }
 }
 
 @immutable
 abstract class Projection {
-  const Projection();
-
-  Bounds<double>? get bounds;
-
-  Point<double> project(LatLng latlng);
-
-  LatLng unproject(Point point);
+  final Bounds<double>? bounds;
 
-  double _inclusive(double start, double end, double value) {
-    if (value < start) return start;
-    if (value > end) return end;
+  const Projection(this.bounds);
 
-    return value;
+  Point<double> project(LatLng latlng) {
```

Comment:
Does it ever make sense to override this? Maybe `@nonVirtual` as well?

Reply:
Done. Crossing my fingers that nobody overwrote this :)

Reply:
Have I missed something? Shouldn't people just override `projectXY`?